src/pybind/mgr/dashboard/cherrypy_backports.py

```diff
@@ -118,8 +150,54 @@ def wrapper(self, sock):
         patch_builtin_ssl_wrap(v, accept_socket_error_0)
 
 
+def patch_request_unique_id(v):
+    """
+    Older versions of cherrypy don't include request.unique_id field (a lazily
+    calculated UUID4).
+
+    Monkey-patching is preferred over alternatives as inheritance, as it'd break
+    type checks (cherrypy/lib/cgtools.py: `isinstance(obj, _cprequest.Request)`)
+    """
+    if v < StrictVersion('11.1.0'):
+        import uuid
+        from functools import wraps
+        from cherrypy._cprequest import Request
+
+        class LazyUUID4(object):
+            def __str__(self):
+                """Return UUID4 and keep it for future calls."""
+                return str(self.uuid4)
+
+            @property
+            def uuid4(self):
+                """Provide unique id on per-request basis using UUID4.
+                It's evaluated lazily on render.
+                """
+                try:
+                    self._uuid4
+                except AttributeError:
+                    # evaluate on first access
+                    self._uuid4 = uuid.uuid4()
+
+                return self._uuid4
+
+        def post_action(func, action):
+            @wraps(func)
+            def wrapper(*args, **kwargs):
+                ret = func(*args, **kwargs)
+                action(*args, **kwargs)
+                return ret
+            return wrapper
+
+        Request.__init__ = post_action(
+            Request.__init__,
+            lambda self, *args, **kwargs: setattr(self, 'unique_id', LazyUUID4())
```

Comment:
i like the idea of generalizing the `post_action()`, but would it be simpler if we just hardwire `set.unique_id = LazyUUID4()` to the constructor? like
```py
def post_action(func, action):
  @wraps(func)
    def wrapper(self, *args, **kwargs):
      ret = func(self, *args, **kwargs)
      self.unique_id = LazyUUID4()
    return wrapper
```

Reply:
Yeah, I over-complicated it a bit :see_no_evil: Initially I just did:
```py
old_init = Request.__init__

def new_init(self, *args, **kw):
  old_init(self, *args, **kw) # To avoid recursion
  self.unique_id = LazyUUID4()

Request.__init__ = new_init
```
But it looked too hacky to me and that's why I tried the pseudo-decorator approach.

Reply:
okay. i am fine either way. i just like
```py
self.unique_id = LazyUUID4()
```
more than
```py
lambda self, *args, **kwargs: setattr(self, 'unique_id', LazyUUID4())
```